server/src/test/java/org/candlepin/resource/ConsumerResourceUpdateTest.java

```diff
@@ -815,6 +815,33 @@ public void consumerCapabilityUpdate() {
     }
 
     @Test
+    public void consumerChangeDetection() {
+        Consumer existing = getFakeConsumer();
+        Set<ConsumerCapability> caps1 = new HashSet<ConsumerCapability>();
+        Set<ConsumerCapability> caps2 = new HashSet<ConsumerCapability>();
+        ConsumerCapability cca = new ConsumerCapability(existing, "capability_a");
+        ConsumerCapability ccb = new ConsumerCapability(existing, "capability_b");
+        ConsumerCapability ccc = new ConsumerCapability(existing, "capability_c");
+        caps1.add(cca);
+        caps1.add(ccb);
+        caps1.add(ccc);
+        caps2.add(ccb);
+
+        existing.setCapabilities(caps1);
+
+        Consumer toUpdate = getFakeConsumer();
+        toUpdate.setCapabilities(caps1);
+        assertFalse(resource.performConsumerUpdates(existing, toUpdate, null));
+
+        toUpdate.setCapabilities(caps2);
+        assertTrue(resource.performConsumerUpdates(existing, toUpdate, null));
+
+        toUpdate.setCapabilities(null);
```

Comment:
Consumer.setCapabilities checks that null is never set into the Consumer.capabilities field. I think intention of this part of the test was to test this path: 
```java
 if (update.getCapabilities() != null) {
```
it doesn't do that.